spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/AbstractLocalDeployerSupport.java

```diff
@@ -194,45 +201,61 @@ protected ProcessBuilder buildProcessBuilder(AppDeploymentRequest request, Map<S
 		}
 
 		ProcessBuilder builder = new ProcessBuilder(commands);
+
 		if (!(request.getResource() instanceof DockerResource)) {
-			builder.environment().putAll(appInstanceEnv);
-			builder.environment().putAll(appInstanceEnvToUse);
+			builder.environment().putAll(appPropertiesToUse);
 		}
+
 		retainEnvVars(builder.environment().keySet());
 
 		if (this.containsValidDebugPort(request.getDeploymentProperties(), deploymentId)) {
-			int portToUse = calculateDebugPort(request.getDeploymentProperties(), appInstanceNumber.orElseGet(() -> 0));
-			builder.command().add(1, this.buildRemoteDebugInstruction(
+
+			int portToUse = calculateDebugPort(request.getDeploymentProperties(), appInstanceNumber.orElse(0));
+
+			String debugInstruction = this.buildRemoteDebugInstruction(
 					request.getDeploymentProperties(),
 					deploymentId,
-					appInstanceNumber.orElseGet(() -> 0),
-					portToUse));
+					appInstanceNumber.orElse(0),
+					portToUse);
+
+			if(request.getResource() instanceof DockerResource) {
+				builder.command().add(2, "-e");
+				builder.command().add(3, "JAVA_TOOL_OPTIONS=\""+ debugInstruction + "\"");			}
```

Comment:
Is this `JAVA_TOOL_OPTIONS` working for Docker resource?

Reply:
No.  Your other PR should address this functionality.